File: 1945-sum-of-digits-of-string-after-convert/1945. Sum of Digits of String After Convert.java
class Solution {
    public int getLucky(String s, int k) {
            int answer = 0;
            StringBuilder sb = new StringBuilder();
            for(int i = 0; i <s.length(); i++) {
                int j = s.charAt(i)-'a'+1;
                sb.append(j);
            }

            while(k>0){
                int sum = 0;
                for(int i = 0; i<sb.length(); i++){
                    sum += sb.charAt(i)-'0';
                }
                sb.setLength(0);
                sb.append(sum);
                answer = sum;


                k--;
            }
            return answer;
    }
}

File: 1945-sum-of-digits-of-string-after-convert/1945. Sum of Digits of String After Convert.cpp
class Solution {
public:
    int getLucky(string s, int k) {
        
        vector<int> v;
        int num,sum=0,tsum=0;
        for(int i=0;i<s.length();i++)
        {                                       // -->(1)
            num =  int(s[i]-96);
            if(num>=10)
            {
                while(num>0)
                {
                    v.push_back(num%10);
                    num/=10;
                }
            }
            else
                v.push_back(num);
        }
        for(int i=0;i<v.size();i++)
            sum+=v[i];                          // -->(2)

        while(k>1)
        {                                       // -->(3)
            tsum=0;
            while(sum>0)
            {                                   // -->(4)
                tsum+=sum%10;
                sum/=10;
            }
            sum=tsum;
            k--;
        }
        return sum;                             // -->(5)
    }
};